import "dccl/option_extensions.proto";
import "dccl/arithmetic/protobuf/arithmetic_extensions.proto";

enum Enum1
{
    ENUM_A = 1;
    ENUM_B = 2;
    ENUM_C = 3;
}

enum Enum2
{
    ENUM2_A = 1;
    ENUM2_B = 2;
    ENUM2_C = 3;
    ENUM2_D = 4;
    ENUM2_E = 5;
}

message ArithmeticDoubleTestMsg
{
    option (dccl.msg).id = 1;
    option (dccl.msg).max_bytes = 512;

    repeated double value = 101 [
        (dccl.field).codec = "_arithmetic",
        (dccl.field).(arithmetic).model = "model",
        (dccl.field).(arithmetic).debug_assert = true,
        (dccl.field).max_repeat = 4
    ];
}
message ArithmeticEnumTestMsg
{
    option (dccl.msg).id = 2;
    option (dccl.msg).max_bytes = 512;

    repeated Enum1 value = 114 [
        (dccl.field).codec = "_arithmetic",
        (dccl.field).(arithmetic).model = "model",
        (dccl.field).(arithmetic).debug_assert = true,
        (dccl.field).max_repeat = 4
    ];
}

message ArithmeticSingleEnumTestMsg
{
    option (dccl.msg).id = 3;
    option (dccl.msg).max_bytes = 512;

    required Enum1 value = 114 [
        (dccl.field).codec = "_arithmetic",
        (dccl.field).(arithmetic).model = "model",
        (dccl.field).(arithmetic).debug_assert = true
    ];
}

message ArithmeticEnum2TestMsg
{
    option (dccl.msg).id = 4;
    option (dccl.msg).max_bytes = 512;

    repeated Enum2 value = 114 [
        (dccl.field).codec = "_arithmetic",
        (dccl.field).(arithmetic).model = "model",
        (dccl.field).(arithmetic).debug_assert = true,
        (dccl.field).max_repeat = 8
    ];
}

message ArithmeticDouble2TestMsg
{
    option (dccl.msg).id = 5;
    option (dccl.msg).max_bytes = 10000;

    repeated int32 value = 101 [
        (dccl.field).codec = "_arithmetic",
        (dccl.field).(arithmetic).model = "model",
        (dccl.field).(arithmetic).debug_assert = true,
        (dccl.field).max_repeat = 100
    ];
}

message ArithmeticDouble3TestMsg
{
    option (dccl.msg).id = 6;
    option (dccl.msg).max_bytes = 10000;

    repeated int32 value = 101 [
        (dccl.field).codec = "_arithmetic",
        (dccl.field).(arithmetic).model = "model",
        (dccl.field).max_repeat = 4,
        (dccl.field).(arithmetic).debug_assert = true
    ];
}

// repeated float float_arithmetic_repeat = 102 [(dccl.field).(arithmetic).model
// = "float_model",
//                                              (dccl.field).max_repeat=4];

// repeated int32 int32_arithmetic_repeat = 103 [(dccl.field).(arithmetic).model
// = "int_model",
//                                              (dccl.field).max_repeat=4];

// repeated bool bool_arithmetic_repeat = 113 [(dccl.field).(arithmetic).model =
// "bool_model",
//                                             (dccl.field).max_repeat=4];
